### DISCUSSION TITLE
Duda de la Kata 7, no me funciona.

### DISCUSSION THREAD

**Discussion by IrmaDiazA** [LaunchX-InnovaccionVirtual/CursoIntroPython#236] [Q&A] [answered]
![problemakata7](https://user-images.githubusercontent.com/99142660/153918396-9bee928d-1e37-40b6-b3e1-b674a0272c22.jpg)
Buen día, tengo un problema al querer darle run al código de la kata 7, simplemente no me funciona, se queda corriendo y no pasa absolutamente nada. ¿Alguien sabe qué podría estar sucediendo?

**Comment by SebbyLebby** [comment]
Hola! Cuando entrás al bucle, lo que hace primero es irse con tu condición 'if', de ahí se agrega el nuevo planeta; pero debes salirte del 'if' para ingresar new_planet = input(...) porque si no, ahí se quedará atorado, ya que su valor no podrá cambiar al estar atrapado la otra instrucción ("new_planet = input(...)) "en ese 'if'.
Espero haya sido de ayuda y no haberte confundido. Sigue intentando, tú puedes!

**Comment by Stmark1** [comment]
No es un problema, el comando input hace que en la parte superior de vscode aparesca un rectangulo para que escribas, poniendo el texto y dandole enter es como se van afregandose. Ahora, para terminar el bucle while, debes escribir _done_ en el mismo rectangulo.

**Comment by KevinBernardoA** [comment]
Antes del if, debes indicarle al programa que vas a darle valor de entrada de usuario a la variable new_planet con la siguente linea: new_planet = input(' ') ahí le das valor ya sea un planeta o la palabra done, si escribes un planeta entra al bucle while, si escribes done se finaliza el programa por que se cumple la condición while que indicaste.

**Comment by doguedogue** [comment]
La línea del input tiene una indentación/tab extra, es decir ahora la tienes dentro del if, solo bájala un nivel para que sea parte del ciclo al nivel del if.

![image](https://user-images.githubusercontent.com/23409026/153927273-5d52a07c-f3d2-45a9-99d0-9cd4cca00a05.png)

**Comment by isaaclc98** [comment, marked as answer]
Como lo menciona @doguedogue , la linea donde pides datos debe de ir fuera del bloque if (solo borra el espacio de la derecha), además recuerda que al poner un input el programa esperara a que lo ingreses vía teclado, aparecerá en la parte de arriba un "cuadro de texto" donde puedes escribir, escribe datos y dale enter para continuar. para finalizar el programa escribe "done".

![Captura](https://user-images.githubusercontent.com/37425783/153930820-5ec3c0a3-fcd5-4d32-bdd6-48ee05569db7.PNG)

**Comment by Jose-Galicia** [comment]
Dentro del ciclo while tiene la funcion if, en esta le estas pidiendo que si existe algun valor en la variable newplanet agregue ese dato a la lista planets y en caso contrario te pide que el usuario ingrese un dato.
Lo que tienes en tu codigo es que si existe un dato se lo pidas al usuario dentro del if, realmente debe de ser fuera porque al inicar el programa no tienes ningun dato en esa variable asi que no entra a la funcion if.
![image](https://user-images.githubusercontent.com/99226240/153933626-ea7e1629-2b2b-4499-9a16-827233e09378.png)

**Comment by IrmaDiazA** [comment]
Muchas gracias a todos, ya pude :D
